### PR TITLE
feat(sdk): split org mgmt create and update

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#438

### Dependencies

- Upstream: none
- Downstream: https://github.com/metriport/metriport-internal/pull/441

### Description

BREAKING CHANGE:
- split org mgmt createOrUpdate in individual create and update endpoints

### Release Plan

- anytime along dependencies
- alpha SDK already deployed

### CODE REVIEW

**Review by Orta21 — APPROVED** [review]
🚢

File: api/app/src/routes/medical/organization.ts [hunk 1]
```diff
@@ -5,50 +5,62 @@ import { createOrganization } from "../../command/medical/organization/create-or
...
-// TODO split this in two, one to create "POST /" and another to update "POST /:id"
 /** ---------------------------------------------------------------------------
  * POST /organization
  *
- * Updates or creates the organization if it doesn't exist already.
+ * Creates a new organization at Metroport and HIEs.
  *
- * @return  {OrganizationDTO}  The organization.
+ * @param req.body The data to create the organization.
+ * @return The newly created organization.
```
Comment (Goncharo) [inline api/app/src/routes/medical/organization.ts:20]: nit: returns

File: api/app/src/routes/medical/organization.ts [hunk 2]
```diff
@@ -5,50 +5,62 @@ import { createOrganization } from "../../command/medical/organization/create-or
...
-        addressLine2: reqOrgData.location.addressLine2 ?? undefined,
-      },
-    };
+/** ---------------------------------------------------------------------------
+ * PUT /organization/:id
+ *
+ * Updates the organization at Metriport and HIEs.
+ *
+ * @param req.body The data to udpate the organization.
+ * @return The updated organization.
```
Comment (Goncharo) [inline api/app/src/routes/medical/organization.ts:46]: nit: returns

File: packages/packages/api/src/medical/client/metriport.ts [hunk 3]
```diff
@@ -48,15 +48,32 @@ export class MetriportMedicalApi {
...
+    const resp = await this.api.post(this.ORGANIZATION_URL, data);
+    if (!resp.data) throw new Error(`Did not receive an organization from the server`);
+    return organizationSchema.parse(resp.data);
+  }
+
+  /**
+   * Updates an organization.
+   *
+   * @param organization The organization data to be updated.
+   * @return The updated organization.
```
Comment (Goncharo) [inline packages/packages/api/src/medical/client/metriport.ts:66]: nit: returns

**Review by Goncharo — APPROVED** [review]
tiny nits, otherwise 🚢 🚢 🚢

**Comment by leite08** [comment]
@Goncharo @Orta21 I changed the `GET /organization` endpoint so it 404s if no org is found, cool?

**Review by Goncharo — APPROVED** [review]
🚢 🚢 🚢

**Pull request merged**